### PR TITLE
GS-HW: Avoid clears with new targets

### LINKED ISSUE #6604: [BUG]: Front Mission 4 broken shadows using any hardware renderer
### Describe the Bug

When using any of the hardware rendering options the shadows in Front Mission 4 are opaque and much larger than they should be. Happens on both my Nvidia and AMD graphics. 
pcsx2-qtx64-avx2_3Q5cJev8NZ
 

### Reproduction Steps

Can be found in the opening in engine cutscene but is present throughout the entire game. 

### Expected Behavior

Shadows should be much smaller with a "softer" and transparent appearance like they are when using the software renderer. 
pcsx2-qtx64-avx2_nodWG0RjCt
 

### PCSX2 Revision

1.7.3066

### Operating System

Windows 10 (64bit)

### If Linux - Specify Distro

_No response_

### CPU

Ryzen 5900HX

### GPU

3060 Mobile 

### GS Settings

_No response_

### Emulation Settings

_No response_

### GS Window Screenshots

_No response_

### Logs & Dumps

_No response_

### LINKED ISSUE #8645: [BUG]: Psychonauts distortion on right half of the screen in certain areas
### Describe the Bug

There's this weird after image or distorted ghosting on right side of the screen in this area of the game. Not sure if it occurs in other areas but I imagine it does at some point. 

### Reproduction Steps

I guess get to this area of the game or load the state or gs dump.

### Expected Behavior

No distortion 

### PCSX2 Revision

v1.7.4372

### Operating System

Windows 11

### If Linux - Specify Distro

_No response_

### CPU

i7 8086k

### GPU

Nvidia GTX 1080ti

### GS Settings

Default, looks fine in sw.

### Emulation Settings

default

### GS Window Screenshots

Psychonauts_SLUS-21120_20230411204431

### Logs & Dumps

GS dump:
[snaps.zip](https://github.com/PCSX2/pcsx2/files/11206995/snaps.zip)

State:
[SLUS-21120 (2373FD16).01.p2s.zip](https://github.com/PCSX2/pcsx2/files/11206997/SLUS-21120.2373FD16.01.p2s.zip)

### LINKED ISSUE #8860: [BUG]: Armored Core 2 FMVs have issues while using hardware rendering.
### Describe the Bug

Some game FMVS will have issues drawing the image, displaying scanlines on top of the screen and making the characters to have an epileptic border and textures while using vulkan or opengl renders, not matter if you upscale or not, the only way to make it go away is to switch from one hardware render to another or switching to software, were this issues doesn't occur 
(I haven't been able to test if this issue happens on the ntsc version due to me not having yet a copy for the game, so if anybody can test if that happens in that version too please told us!)
Screenshot from 2023-05-13 01-09-20
Screenshot from 2023-05-14 22-22-33
Screenshot from 2023-05-13 01-13-24

### Reproduction Steps

Start up the game with any hardware rendering, the FMV after the "from software" logo will display the issue.

### Expected Behavior

Imagine rendering without any image corruption on the characters and without any scan-lines on top 
Software mode
Screenshot from 2023-05-13 01-32-59 
Hardware render if you change to another hardware render in the middle of the cutscene
Screenshot from 2023-05-14 22-31-56

### PCSX2 Revision

v1.7.4508

### Operating System

Linux (64bit) - Specify distro below

### If Linux - Specify Distro

EndeavourOS (Arch Linux based distro)

### CPU

Ryzen 5 5600

### GPU

Rx 6600

### GS Settings

_No response_

### Emulation Settings

Screenshot from 2023-05-14 22-36-42

### GS Window Screenshots

_No response_

### Logs & Dumps

Armored Core 2_SLES-50079_20230513010926
[Armored Core 2_SLES-50079_20230513010926.gs.zip](https://github.com/PCSX2/pcsx2/files/11473914/Armored.Core.2_SLES-50079_20230513010926.gs.zip)

### PR DESCRIPTION
**Pull request by refractionpcsx2**
### Description of Changes
Attempts to avoid clearing targets which don't already exist

### Rationale behind Changes
Before this would create errant targets, sometimes located in the middle of buffers (Armored Core 2), which would later cause misdetection when it actually required those addresses.  This was also causing some blowout of target sizes when it wasn't needed.

### Suggested Testing Steps
Test a whole bunch of games, and especially FMV's

Fixes #8645 Psychonauts weird top screen corruption
Fixes #6604 Front Mission 4 shadows
Fixes #8860 Armored Core 2 FMV flickering
Improves accuracy of Time Crisis 3 menu
Improves accuracy on Cricket 2002 camera view

Psychonauts:
Master:
image

PR:
image

Front Mission 4:
Master:
image

PR:
image

Cricket 2002:
Master:
image

PR:
image

Time Crisis 3: (Comapre the honeycomb effect at the bottom centre of the screen, more accurately matches software mode)
Master:
image

PR:
image

### CODE REVIEW

File: pcsx2/GS/Renderers/HW/GSTextureCache.cpp [hunk 1]
```diff
@@ -2776,13 +2778,15 @@ bool GSTextureCache::ShuffleMove(u32 BP, u32 BW, u32 PSM, int sx, int sy, int dx
...
 
-GSTextureCache::Target* GSTextureCache::GetExactTarget(u32 BP, u32 BW, int type)
+GSTextureCache::Target* GSTextureCache::GetExactTarget(u32 BP, u32 BW, int type, u32 end_bp)
 {
 	auto& rts = m_dst[type];
 	for (auto it = rts.begin(); it != rts.end(); ++it) // Iterate targets from MRU to LRU.
 	{
 		Target* t = *it;
-		if (t->m_TEX0.TBP0 == BP && t->m_TEX0.TBW == BW)
+		const u32 end_block = (t->m_end_block < t->m_TEX0.TBP0) ? t->m_end_block + 0x4000 : t->m_end_block;
```
Comment (stenzek) [inline pcsx2/GS/Renderers/HW/GSTextureCache.cpp:2787]:
```suggestion
		const u32 end_block = t->UnwrappedEndBlock();
```

**Comment by AmyRoxwell** [comment]
This PR does help with the issue on armored core 2, but there is another bug caused after it is fixed, if you use hardware mode there will like 1 full line of garbage tiles been drawn at the bottom of the screen in all FMVs, even some of them will have garbage at the right side.
Hardware With PR:
Screenshot from 2023-05-17 20-08-00
Screenshot from 2023-05-17 20-26-59
Screenshot from 2023-05-17 20-28-03
Software mode:
Screenshot from 2023-05-17 20-27-02
Screenshot from 2023-05-17 20-30-02
Screenshot from 2023-05-17 20-30-35

**Comment by refractionpcsx2** [comment]
That only happens if you upscale past x2, this is not uncommon and I won't be dealing with it. You will need to crop the bottom by 1px in the options.

**Comment by AmyRoxwell** [comment]
Alrighty, thanks for the answer!

**Review by Mrlinkwii — APPROVED** [review]

**Review by DigitalMajestic — APPROVED** [review]

**Review by JordanTheToaster — APPROVED** [review]

**Comment by OCRBonk** [comment]
Fantastic! I have been waiting over the years hoping someone at some point would figure out a fix for the shadows in FM4. Time for a playthrough soon. Thank you!

**Pull request merged**